homeassistant/components/zwave_js/__init__.py

```diff
@@ -313,19 +313,22 @@ async def async_on_node_added(self, node: ZwaveNode) -> None:
                 node,
             )
 
-        # we only want to run discovery when the node has reached ready state,
-        # otherwise we'll have all kinds of missing info issues.
-        if node.ready:
-            await self.node_events.async_on_node_ready(node)
-            return
-        # if node is not yet ready, register one-time callback for ready state
-        LOGGER.debug("Node added: %s - waiting for it to become ready", node.node_id)
-        node.once(
+        LOGGER.debug("Node added: %s", node.node_id)
+
+        # Listen for ready node events, both new and re-interview.
+        node.on(
```

Comment:
Should we remove this listener when ready fires?

Reply:
Ah read the description better 👍

Reply:
We should remove it on config entry unload though. I'll add that.